hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheRecordStore.java

```diff
@@ -1052,6 +1138,66 @@ public Object get(Data key, ExpiryPolicy expiryPolicy) {
         }
     }
 
+    public void evictExpiredEntries(int expirationPercentage) {
+        long now = Clock.currentTimeMillis();
+        int maxIterationCount = getMaxIterationCount(size(), expirationPercentage);
+        int evictedCount = 0;
+        int maxRetry = 3;
+        int loop = 0;
+        while (true) {
+            evictedCount += evictExpiredInternal(maxIterationCount, now);
+            if (evictedCount >= maxIterationCount) {
+                break;
+            }
+            loop++;
+            if (loop > maxRetry) {
+                break;
+            }
+        }
+    }
+
+    protected void initExpirationIterator() {
+        if (expirationIterator == null || !expirationIterator.hasNext()) {
+            expirationIterator = records.entrySet().iterator();
+        }
+    }
+
+    private int evictExpiredInternal(int maxIterationCount, long now) {
+        initExpirationIterator();
+        int processedCount = 0;
+        LinkedList<Map.Entry<Data, R>> records = new LinkedList<Map.Entry<Data, R>>();
+        while (expirationIterator.hasNext()) {
```

Comment:
`while (expirationIterator.hasNext() && processedCount < maxIterationCount)` could get rid of the extra `if break` condition.